drivers/kw2xrf/kw2xrf_intern.c

```diff
@@ -0,0 +1,230 @@
+/*
+ * Copyright (C) 2016 PHYTEC Messtechnik GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup     drivers_kw2xrf
+ * @{
+ * @file
+ * @brief       Internal function of kw2xrf driver
+ *
+ * @author      Johann Fischer <j.fischer@phytec.de>
+ * @}
+ */
+
+#include "panic.h"
+#include "kw2xrf.h"
+#include "kw2xrf_spi.h"
+#include "kw2xrf_reg.h"
+#include "kw2xrf_getset.h"
+#include "kw2xrf_intern.h"
+#include "overwrites.h"
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+void kw2xrf_disable_interrupts(kw2xrf_t *dev)
+{
+    /* Clear and disable all interrupts */
+    kw2xrf_write_dreg(dev, MKW2XDM_PHY_CTRL2, 0xff);
+    int reg = kw2xrf_read_dreg(dev, MKW2XDM_PHY_CTRL3);
+    reg |= MKW2XDM_PHY_CTRL3_WAKE_MSK | MKW2XDM_PHY_CTRL3_PB_ERR_MSK;
```

Comment:
mhm, maybe a call is missing, in the current driver (`kw2xrf.c`) I found:

```
        /* Mask all possible interrupts */
        reg = kw2xrf_read_dreg(MKW2XDM_PHY_CTRL3);
        reg |= MKW2XDM_PHY_CTRL3_WAKE_MSK;
        kw2xrf_write_dreg(MKW2XDM_PHY_CTRL3, reg);

        kw2xrf_write_dreg(MKW2XDM_IRQSTS1, 0x7f);
        kw2xrf_write_dreg(MKW2XDM_IRQSTS2, 0x03);
        kw2xrf_write_dreg(MKW2XDM_IRQSTS3, 0xff);
```

so `reg` is written back to the register, that is missing here?!